52.md

```diff
@@ -55,13 +58,18 @@ The following tags are deprecated:
     ["end", "<YYYY-MM-DD>"],
 
     // Location
-    ["location", "<location>"],
+    ["address", "<street name and number>", "<city or locality>", "<postal or zip code>", "<country>"],
+    ["place", "<place>"],
     ["g", "<geohash>"],
+    ["url", "<nostr:naddr1xxx or a http(s)://xxx>"],
 
     // Participants
     ["p", "<32-bytes hex of a pubkey>", "<optional recommended relay URL>", "<role>"],
     ["p", "<32-bytes hex of a pubkey>", "<optional recommended relay URL>", "<role>"],
 
+    // Languages
+    ["languages", "<lang>", "<lang>"]
```

Comment:
Is `l` standardized yet? Multiple `l` tags might be better.

Reply:
I searched the NIPs but found no strong evidence on `l`; however, I am fine with that.
Why is it better to repeat? Repetition is necessary if the data is structured (`e`, `p`, etc.) and might be useful if it could become structured in the future, otherwise a list is more compact. Could languages have structure? Perhaps an optional `primary` or `secondary` field.

Reply:
Repeating allows relays to index the language value and clients to search by it

Reply:
Single-letter tags are indexable and searchable. So it can be easily queried.

Introducing `primary` and `secondary` seem to add additional complexity that doesn't seem necessary, and nobody has really asked for that kind of functionality yet. Folks can always use the `content` field to describe in greater detail if needed.

Someone just introduced a language tag proposal: https://github.com/nostr-protocol/nips/pull/1127
@alexgleason also introduced one last year but it didn't get merged: https://github.com/nostr-protocol/nips/pull/632

I think there's two distinct concepts around language, although we may or may not be able to merge the concepts into the same tag (not sure):
1. Language of the Nostr event (i.e. the contents of the payload are written in this language and can be more reliably translated to a different language)
2. Language of the **calendar event** (i.e. if I attend this onsite or online event, I know that the people there will be using this language)

Reply:
@tyiu I was not proposing to make the language tag more complex _now_, I was just speculating why a repeated tag could be useful in a future perspective. _Indexable and searchable_ is definitively the answer.